src/test/java/uk/ac/ebi/eva/contigalias/controller/contigalias/ContigAliasHandlerTest.java

```diff
@@ -230,21 +231,54 @@ void assertChromosomeIdenticalToEntity(ChromosomeEntity chromosome) {
 
         @BeforeEach
         void setup() {
-            AliasService mockAliasService = mock(AliasService.class);
+            ChromosomeService mockChromosomeService = mock(ChromosomeService.class);
             for (int i = 0; i < CHROMOSOME_LIST_SIZE; i++) {
                 ChromosomeEntity generate = ChromosomeGenerator.generate(i, assemblyEntity);
                 chromosomeEntities.add(generate);
                 Optional<AssemblyEntity> entityOptional = Optional.of(this.assemblyEntity);
-                Mockito.when(mockAliasService.getAssemblyByChromosomeGenbank(generate.getGenbank()))
+                Mockito.when(mockChromosomeService.getAssemblyByChromosomeGenbank(generate.getGenbank()))
                        .thenReturn(entityOptional);
-                Mockito.when(mockAliasService.getAssemblyByChromosomeRefseq(generate.getRefseq()))
+                Mockito.when(mockChromosomeService.getAssemblyByChromosomeRefseq(generate.getRefseq()))
                        .thenReturn(entityOptional);
             }
-            Mockito.when(mockAliasService.getChromosomesByAssemblyGenbank(assemblyEntity.getGenbank()))
+            Mockito.when(mockChromosomeService.getChromosomesByAssemblyGenbank(assemblyEntity.getGenbank()))
                    .thenReturn(chromosomeEntities);
-            Mockito.when(mockAliasService.getChromosomesByAssemblyRefseq(assemblyEntity.getRefseq()))
+            Mockito.when(mockChromosomeService.getChromosomesByAssemblyRefseq(assemblyEntity.getRefseq()))
                    .thenReturn(chromosomeEntities);
-            handler = new ContigAliasHandler(null, null, mockAliasService, null, null);
+            String chrName = chromosomeEntities.get(0)
+                                               .getName();
+            Long asmTaxid = assemblyEntity.getTaxid();
+            Mockito.when(
+                    mockChromosomeService.getChromosomesByNameAndAssemblyTaxid(chrName, asmTaxid, DEFAULT_PAGE_REQUEST))
+                   .thenReturn(new PageImpl<>(chromosomeEntities
+                                                      .parallelStream()
+                                                      .filter(it -> it.getName().equals(chrName) &&
+                                                              it.getAssembly().getTaxid().equals(asmTaxid))
+                                                      .collect(Collectors.toList())));
+
+            List<ChromosomeEntity> chromosomesByNameAndAssembly = chromosomeEntities
+                    .parallelStream()
```

Comment:
why parallel stream? speed? have you measured it?

Reply:
I haven't measured it. Parallelized it for no good reason. Have removed it now.